src/lib/forms/core/StringField.js

```diff
@@ -15,13 +16,32 @@ export class StringField extends Component {
   }
 
   renderFormField = (props) => {
-    const { fieldPath, inline, width, optimized, ...uiProps } = this.props;
+    const {
+      fieldPath,
+      inline,
+      width,
+      optimized,
+      popupHelper,
+      label,
+      ...uiProps
+    } = this.props;
     const {
       form: { values, handleChange, handleBlur, errors, status, isSubmitting },
     } = props;
-
+    const { required } = uiProps;
+    let popupLabelClassName = 'display-flex';
+    if (popupHelper && required) {
+      popupLabelClassName += ' required field';
+    }
     return (
       <Form.Field inline={inline} width={width} disabled={isSubmitting}>
+        {popupHelper ? (
```

Comment:
We are going to specific here IMHO: some fields have a popup, some other a label.
We are abusing of generic components and hardcoding a specific layout of a specific form. At this point, it is better to created a specific `DocReqStringField` that wraps the original/generic one and adds functionality.

A better option is probably to simply pass a `component` prop, and you pass the entire label or popup from the caller.

Remember that the API interface of your cmp/function/method has to be the purest possible and you have to maximise re-usability. The inner implementation should know nothing of the caller or in which context it will be used.